### PR TITLE
[stable/dex] Fix ConfigMap creation and add missed config parameter

### PR DESCRIPTION
**Pull request by lohmag**
1) Fix for openssl config which used by Job later when hook executed. Becouse of `"helm.sh/hook-delete-policy": hook-succeeded` Configmap deletes just after creates.
2) Added ability to set token `expire` which is realised in dex app config file but not in helm `values.yaml`
https://github.com/dexidp/dex/blob/master/examples/config-dev.yaml#L38

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x ] Chart Version bumped
- [x ] Variables are documented in the README.md
- [x ] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by zanhsieh** [comment]
@lohmag 
Would you mind to sign DCO please?

**Comment by vi7** [comment]
hi @lohmag , thanks for the contribution. Could you please squash commits into one and name it according to the introduced changes? Also please put a meaningful name for the PR (can be the same as for commit) and prefix it with `[stable/dex]`

**Comment by lohmag** [comment]
I don't know how to squash commit into one and can't spend anymore time on it. And actually I don't know why I should do it if you can squash in web interface when it merges by just checking a box.

**Comment by desaintmartin** [comment]
With this change, we'll have one unused configmap, but it is better than a non-working Chart. Thanks!

Could you bump minor version of this chart instead of patch version since there is a new feature?

/assign

@vi7 FYI, AFAIK, merge bot will squash the commits anyway, so the important thing is the PR title.

**Comment by lohmag** [comment]
> With this change, we'll have one unused configmap, but it is better than a non-working Chart. Thanks!
> 
> Could you bump minor version of this chart instead of patch version since there is a new feature?
> 
> /assign
> 
> @vi7 FYI, AFAIK, merge bot will squash the commits anyway, so the important thing is the PR title.

what do you mean by version bumping? i already bumped helm chart version

File: stable/dex/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: dex
-version: 2.3.2
+version: 2.3.3
```
Comment (desaintmartin) [inline stable/dex/Chart.yaml]:
```suggestion
version: 2.4.0
```

This introduce a new feature, SemVer wants us to bump MINOR version.

**Review by desaintmartin — CHANGES_REQUESTED** [review]
- bump minor version
- Please also document in the README the new value

**Comment by vi7** [comment]
> @vi7 FYI, AFAIK, merge bot will squash the commits anyway, so the important thing is the PR title.

@desaintmartin I'm ok with that in case merge commit will be named after the PR name, cause current commit names doesn't describe introduced changes and moreover have misleading names

**Pull request merged**